### DISCUSSION TITLE
[DISCUSSION]LoadingView에 버튼 스피너 추가 및 적용 기준 제안

### DISCUSSION THREAD

**Discussion by ffalswo2** [Team-Plu/Plu-iOS#58] [Ideas]
## 로딩뷰 방식 기준 제안
LoadingView에 대한 이야기를 하다가 우리 앱에서는 어떻게 로딩뷰를 가져가면 좋을지에 대해 고민해보았습니다. 로딩뷰는 크게 아래와 같이 3가지가 있는 것 같습니다.

- 스켈레톤뷰
- 스피너
- (커스텀 로딩뷰): 이건 현시점에서 불필요할듯

여기서 만약 MVP단계라는 규모에 맞게 시간이 조금 소요되는 스켈레톤뷰를 제외하고 적용한다면 선택지는 아래 2개일 것 같습니다.

- 전체화면 스피너 (기존의 PLUIndicatorView)
- 버튼 스피너 (Button config의 `showsActivityIndicator` 이용)

로딩이 필요한 경우는 보통 네트워크 통신과정에서 생기는 지연시간동안 유저를 붙잡아두기 위함이라고 생각합니다. 이 때 이 과정에서 아래와 같은 기준을 제시합니다.

- **전체 화면이 새로 그려지는 경우**: 기존 `PLUIndicatorView` (앱에서 다수의 GET통신)
- **부분 업데이트가 적용되는 경우 혹은 유저 액션에 대한 강한 피드백이 필요한 경우**: `버튼 스피너`
아래와 같은 공감 버튼의 경우가 해당될 것 같습니다. 그냥 +1 올리고 네트워크 통신은 뒤에 하면 안되나 싶지만, 이렇게 되면 유저는 실제로 업데이트가 반영이 되었다고 생각하지만 네트워크 통신에서 에러가 날 경우 적용이 안되는게 맞기에 이런 곳은 버튼 스피너가 적용되는게 조금 더 좋아보입니다.
또한 로그인 버튼의 경우 로그인 버튼을 눌렀고 프로세스가 진행중이라는 피드백이 강하게 필요해보이기에 버튼 스피너가 적합해보입니다. 아래 GIF참고해주시면 좋을 것 같습니다.

![image](https://github.com/Team-Plu/Plu-iOS/assets/60292150/f50979d4-2c30-48fa-86b2-a1f48d2187c1)
![Simulator Screen Recording - iPhone 13 mini - 2023-12-19 at 17 10 29](https://github.com/Team-Plu/Plu-iOS/assets/60292150/3270c896-85b0-420f-8e20-a2ec11f14fe9)

버튼 스피너의 경우 아래와 같이 간단하게 구현가능합니다.
```swift
func setActivityIndicator(isShow: Bool) {
    var config = self.configuration
    config?.showsActivityIndicator = isShow
    self.configuration = config
}
```

## 참고
https://medium.com/myrealtrip-product/%EC%83%81%ED%99%A9%EC%97%90-%EB%A7%9E%EB%8A%94-%EB%A1%9C%EB%94%A9-%EC%95%A0%EB%8B%88%EB%A9%94%EC%9D%B4%EC%85%98-%EC%A0%81%EC%9A%A9%ED%95%98%EA%B8%B0-2018af51c197

**Comment by kimscastle** [comment]
버튼에 이미지가 없는 상황에서 indicator가 어색하지 않게 보인다면 이방식이 뭔가 더 깔끔해보이긴하네요 하는 방식도 훨씬쉽네요
